### PR TITLE
Add whitelist system

### PR DESCRIPTION
**Pull request by Thesam1798**
### Whitelist System
Hello, I made during the day a whitelist system for the framework.

There are still some features that I would like to add, like the management of the list directly in game and then a permission and role system, which could manage the different sides (medic and cop) or even take the role of license management.

So the whitelist system is a first step for this permission management.

I invite you to read the small documentation I created to understand quickly the system, you can also contact me on discord ; TheSam#0001

PS: there is also a commit to add a generic logger to the whole framework, tell me what you think.

### CODE REVIEW

File: src/Prefabs/MP/Modes/Roleplay/GameMode_Roleplay.et [hunk 1]
```diff
@@ -59,9 +36,6 @@ EL_GameModeRoleplay {
   ScriptedChatEntity ScriptedChatEntity : "{F69BC912AC8236F9}Prefabs/MP/ScriptedChatEntity.et" {
    ID "59751433747F9C02"
   }
-  SCR_GarbageManager GarbageManager {
-   ID "59751433747F9391"
-  }
```
Comment (Thesam1798) [inline src/Prefabs/MP/Modes/Roleplay/GameMode_Roleplay.et:64]: For information, this was changed when opening the file after a forced save, I did not manage to keep these lines from the editor.

File: src/Prefabs/MP/Modes/Roleplay/GameMode_Roleplay.et [hunk 2]
```diff
@@ -9,31 +9,8 @@ EL_GameModeRoleplay {
...
-      m_rPrefab "{DCF980831E880F6A}Prefabs/Characters/Uniforms/Pants_M88.et"
-     }
-    }
-    EL_DefaultCharacterLoadoutSlot "{5A2882A36132BCD3}" {
-     m_eArea ELA_Boots
-     m_pItem EL_DefaultCharacterLoadoutItem "{5A2882A49CA76329}" {
-      m_rPrefab "{C7923961D7235D70}Prefabs/Characters/Footwear/CombatBoots_Soviet_01.et"
-     }
-    }
-   }
```
Comment (Thesam1798) [inline src/Prefabs/MP/Modes/Roleplay/GameMode_Roleplay.et:36]: For information, this was changed when opening the file after a forced save, I did not manage to keep these lines from the editor.

**Comment by Thesam1798** [comment]
@thebonbon Done 😊

File: src/Scripts/Game/Feature/Whitelist/EL_WhitelistComponent.c [hunk 3]
```diff
@@ -0,0 +1,116 @@
+[ComponentEditorProps(category: "EveronLife/Feature/Whitelist", description: "Allows to thread the users who connect to the server.")]
+class EL_WhitelistComponentClass: SCR_RespawnHandlerComponentClass
+{
+}
+
+class EL_WhitelistComponent : SCR_RespawnHandlerComponent
+{
+	private EL_DbContext m_Database;
+	private bool m_WhitelistEnable = false;
```
[the remark this answers is not in the document]
Comment (thebonbon) [reply src/Scripts/Game/Feature/Whitelist/EL_WhitelistComponent.c:9]: Bool is default false.
And m_**b**WhitelistEnable

https://github.com/EveronLife/EveronLife/blob/main/.github/CONVENTIONS.md#type-prefixes

**Review by thebonbon — CHANGES_REQUESTED** [review]

File: src/Scripts/Game/Feature/Whitelist/EL_WhitelistComponent.c (hunk 3 above)
[the remark this answers is not in the document]
Comment (Thesam1798) [reply src/Scripts/Game/Feature/Whitelist/EL_WhitelistComponent.c]: This was not done in many EL files, so I assumed that it did not apply.

In any case it is Fix

**Comment by Thesam1798** [comment]
Updated to bring everything up to date with multiplayer.

The logic remains the same, but some security and checks have been added.

Test on a dedicated server on `22/10/2022` with the package `5AACC14F4040793D`, V0.1.0 (https://reforger.armaplatform.com/workshop/5AACC14F4040793D-Fork-Everonlife-Thesam1798)

**Comment by thebonbon** [comment]
We do need a proper Logger at some point but I'm not sure if we should add a join-whitelist system to the framework. I think this would fit better as a sperarate standalone mod.

**Comment by Thesam1798** [comment]
Alright no problem, so you can close the pull request

**Pull request closed without merging**